### PR TITLE
#919: [typescript .d.ts] Button (closes #919)

### LINKED ISSUE #919: [typescript .d.ts] Button
← #908

## requirements
{describe the new feature}

## specs
{optional: describe technical specs to implement this feature, if not obvious}

## misc
{optional: other useful info}

### PR DESCRIPTION
**Pull request by ascariandrea**
Closes #919

## Test Plan

> These results can be screenshots, query results, or even just pointers to unit tests that successfully passed.

- [ ] Google Chrome
- [ ] Internet Explorer

> An example of partial test is a field displaying 4 possible values. If 3 values are tested, with screenshots, and 1 is not, then it should be mentioned here.

### CODE REVIEW

File: src/button/Button.d.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  type ButtonState = 'ready' | 'not-allowed' | 'processing' | 'error' | 'success';
+  type ButtonType = 'default' | 'primary' | 'positive' | 'negative' | 'flat';
+  type ButtonSize = 'tiny' | 'small' | 'medium';
+
+}
+
+export type ButtonProps = {
+    onClick: (e: React.SyntheticEvent<HTMLDivElement>) => void,
+    buttonState?: ButtonProps.ButtonState,
+    label?:string | object,
```
Comment (FrancescoCioria) [inline src/button/Button.d.ts]: you should replace `object` with `{ [key in buttonState]: string }`

File: src/button/Button.d.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    icon?:string | object,
```
Comment (FrancescoCioria) [inline src/button/Button.d.ts]: you should replace `object` with `{ [key in buttonState]: string }`

File: src/button/Button.d.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    children?: string,
+    type?: ButtonProps.ButtonType,
+    primary?: boolean,
+    flat?: boolean,
+    size?: ButtonProps.ButtonSize,
+    fluid?: boolean,
+    circular?: boolean,
+    textOverflow?: React.ComponentClass<object>, // todo(typo) when translate text overflow, update here its props types
+    style?: object,
```
Comment (FrancescoCioria) [inline src/button/Button.d.ts]: same as other PR: we should use `React.CSSProperties` for `style`

**Comment by ascariandrea** [comment]
Fixed!

**Pull request merged**